std/range/package.d

```diff
@@ -6730,6 +6734,381 @@ if (isForwardRange!Source && hasLength!Source)
     assert(equal(chunks, [[1], [2], [3], [], []]));
 }
 
+/**
+This range iterates over a fixed-sized sliding window (k-mers)
+of size $(D windowSize) of a $(D source) range.
+$(D Source) must be a forward range. $(D windowSize) must be
+greater than zero. For $(D windowSize = 2) it is similar to
+$D(zip(arr, arr.save.dropOne)).
+
+$(D[1,2,3].slides().array == [[1,2], [2,3]])
+
+Params:
+    r = Range from which the minimum will be selected
+    windowSize = Sliding window size (default 2)
+
+See_Also: $(LREF chunks)
+
+Returns: Input range of all sliding windows
```

Comment:
It's always at least a forward range with propagated bidirectionality and conditional random access and slicing, so stating "Input range" here is confusing. I'd prefer if all the capabilities were succinctly documented.